fixed-map-derive/src/unit_variants.rs

```diff
@@ -192,53 +194,15 @@ pub(crate) fn implement(cx: &Ctxt<'_>, en: &DataEnum) -> Result<TokenStream, ()>
             impl<V> #partial_ord_t for Storage<V> where V: #partial_ord_t {
```

Comment:
`PartialOrd` and `Ord` should probably require that the key type implement the same, even though it isn't strictly necessary for the implementation.

You can achieve this by adding the following additional bound to the where clause (hack for trivial bounds in where clauses):
```
#ident: for<#lt> #partial_ord_t,
```

Reply:
Well, I don't think it's a *great* idea since it gives the wrong impression about ordering: that you can implement PartialOrd to decide how something is ordered when that isn't possible.

It's better to indicate to the user of the library that the `Key` trait comes with its own ordering guarantees. Or in the case where it's backed by a `HashMap` simply isn't ordered (but this would have to be solved later).

Reply:
As a (maybe temporary) compromise: how derived ordering semantics should at least be documented on the `Key` trait and linked to in the collection's `PartialOrd` and `Ord` impls.